File: data/code_pairs/cpp/2077.cpp
#include <iostream>

int fact(int n) {
    if (n <= 1) return 1;
    else return n * fact(n - 1);
}

void test_fact(int test_value, int expected_value, const std::string& test_case_label) {
    int result = fact(test_value);
    if (result == expected_value) {
        std::cout << "PASS: " << test_case_label << std::endl;
    } else {
        std::cout << "FAIL: " << test_case_label 
                  << ". Expected: " << expected_value << ", Got: " << result << std::endl;
    }
}

int main() {
    // Test case 1: Fact(5)
    test_fact(5, 120, "Fact(5)");

    // Test case 2: Fact(0)
    test_fact(0, 1, "Fact(0)");

    // Add more test cases as needed

    return 0;
}

File: data/code_pairs/fortran/2077.f90
program factorial_test
    implicit none

    call test_fact()

contains

    function fact(n) result(f)
        integer, intent(in) :: n
        integer :: f

        if (n <= 1) then
            f = 1
        else
            f = n * fact(n - 1)
        endif
    end function fact

    subroutine test_fact()
        integer :: expected, computed

        ! Test case 1: Fact(5)
        expected = 120
        computed = fact(5)
        call assert_equals(computed, expected, "Fact(5)")

        ! Test case 2: Fact(0)
        expected = 1
        computed = fact(0)
        call assert_equals(computed, expected, "Fact(0)")

        ! Add more test cases as needed
    end subroutine test_fact

    subroutine assert_equals(computed, expected, test_label)
        integer, intent(in) :: computed, expected
        character(len=*), intent(in) :: test_label

        if (computed == expected) then
            print *, "PASS: ", test_label
        else
            print *, "FAIL: ", test_label, " Expected: ", expected, ", Got: ", computed
        endif
    end subroutine assert_equals

end program factorial_test